### PR TITLE
Resolve the problem of json crash when text is not string #6

### LINKED ISSUE #6: Terraform crashes if json output contains non strings
Examples:
```go
data "shell_script" "test" {
  lifecycle_commands {
    read = <<EOF
      echo '{"commit_id": true}' >&3
    EOF
  }
}
```
```go
data "shell_script" "test" {
  lifecycle_commands {
    read = <<EOF
      echo '{"commit_id": 42}' >&3
    EOF
  }
}
```

### PR DESCRIPTION
**Pull request by FredPi17**
closes #6

### CODE REVIEW

File: shell/utility.go [hunk 1]
```diff
@@ -50,7 +49,11 @@ func parseJSON(b []byte) (map[string]string, error) {
 	err := json.Unmarshal([]byte(s), &f)
 	output := make(map[string]string)
 	for k, v := range f {
-		output[k] = v.(string)
```
Comment (ppff) [inline shell/utility.go:53]: idea: `output[k] = fmt.Sprint(v.(string))`

File: shell/utility.go [hunk 2]
```diff
@@ -4,13 +4,12 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"github.com/armon/circbuf"
```
Comment (ppff) [inline shell/utility.go:7]: unrelated to this MR

File: examples/test.tf [hunk 3]
```diff
@@ -99,7 +126,7 @@ output "commit_id2" {
 resource "shell_script" "test6" {
   lifecycle_commands {
     create = file("${path.module}/scripts/create.sh")
-    read   = file("${path.module}/scripts/read.sh")
+    read = file("${path.module}/scripts/read.sh")
```
Comment (ppff) [inline examples/test.tf]: unrelated to this MR

File: examples/test.tf [hunk 4]
```diff
@@ -78,7 +105,7 @@ resource "shell_script" "test4" {
 resource "shell_script" "test5" {
   lifecycle_commands {
     create = file("${path.module}/scripts/create.sh")
-    read   = file("${path.module}/scripts/read.sh")
+    read = file("${path.module}/scripts/read.sh")
```
Comment (ppff) [inline examples/test.tf]: unrelated to this MR

**Comment by ppff** [comment]
About tests:

- [x] remove commented test

- [x] leave existing tests as is (first test should not be modified)

- [x] add test which covers all possible scalar values:
    * string
    * int
    * float
    * bool
    * null

File: shell/utility.go (hunk 2 above)
Comment (FredPi17) [reply shell/utility.go:7]: It's like this because importing is sorted automatically by the plugin

**Comment by ppff** [comment]
Seems better here https://github.com/scottwinkler/terraform-provider-shell/pull/33
¯\\_(ツ)_/¯

**Pull request closed without merging**